src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperRunner.cs

```diff
@@ -24,7 +24,10 @@ public class TagHelperRunner
                 executionContext.AllAttributes,
                 executionContext.UniqueId,
                 executionContext.GetChildContentAsync);
-            var tagHelperOutput = new TagHelperOutput(executionContext.TagName, executionContext.HTMLAttributes);
+            var tagHelperOutput = new TagHelperOutput(executionContext.TagName, executionContext.HTMLAttributes)
+            {
+                SelfClosing = executionContext.SelfClosing,
```

Comment:
I don't understand the question, but in the original codes, selfclosing was always 'false', and each taghelper like input could set it true if necessary. In this fix, selfclosing is set based on the original user input tags from cshtml, and individual taghelpers can change it as needed.